### PR TITLE
(Fixes #349) SplitMergeFilesActivity needs to pass `temporary-directory`

### LINKED ISSUE #349: SplitMergeFilesActivity needs to pass `temporary-directory`

### PR DESCRIPTION
**Pull request by sethyates**

### CODE REVIEW

File: core/src/main/scala/com/krux/hyperion/adt/HType.scala [hunk 1]
```diff
@@ -113,7 +113,12 @@ case class HDouble(value: Either[Double, DoubleExp]) extends HType with Optional
 
 }
 
-case class HBoolean(value: Either[Boolean, BooleanExp]) extends HType
+case class HBoolean(value: Either[Boolean, BooleanExp]) extends HType {
+  def exists[B](fn: => B) = value match {
```
Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/adt/HType.scala]: why call by name?

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/adt/HType.scala]: huh?

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/adt/HType.scala]: I mean why use `fn: => B` instead of `fn: B`?

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/adt/HType.scala]: Isn't that typical for these sorts?  I don't want to have `fn` evaluated unless it's actually required.  My understanding is if I passed by value, Scala would evaluate the expression before passing into the function.

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/adt/HType.scala]: I seems not matter in this case. Just curious, I thought there's a particular reason.

**Comment by realstraw** [comment]
:+1: :sparkles: :sparkles: :sparkles:

**Pull request merged**